sdv/constraints/tabular.py

```diff
@@ -59,69 +73,99 @@ class CustomConstraint(Constraint):
             Function to replace the ``reverse_transform`` method.
         is_valid (callable):
             Function to replace the ``is_valid`` method.
+
+    Returns:
+        CustomConstraint class:
+            A constraint with custom ``transform``/``reverse_transform``/``is_valid`` methods.
     """
+    _validate_inputs_custom_constraint(is_valid_fn, transform_fn, reverse_transform_fn)
 
-    def _run(self, function, table_data, reverse=False):
-        table_data = table_data.copy()
-        if self._columns:
-            if reverse:
-                columns = reversed(self._columns)
-            else:
-                columns = self._columns
-
-            for column in columns:
-                try:
-                    table_data = function(table_data, column)
-                except TypeError:
-                    table_data[column] = function(table_data[column])
+    class CustomConstraint(Constraint):
+        """Custom Constraint Class.
 
-        else:
-            table_data = function(table_data)
+        Args:
+            transform (callable):
+                Function to replace the ``transform`` method.
+            reverse_transform (callable):
+                Function to replace the ``reverse_transform`` method.
+            is_valid (callable):
+                Function to replace the ``is_valid`` method.
+        """
 
-        return table_data
+        TYPE = 'CUSTOM_CONSTRAINT'  # this lets us know how to handle it
```

Comment:
Can we either make this a boolean flag like `is_custom` or create an enum class for this? The reason would be that whenever someone is doing logic around this attribute, they would have to know the string value for custom constraints and any other type. It would be easier if they could see all the possible values. Here is an example of enums: https://docs.python.org/3/library/enum.html

Personally though, I think this can just be a boolean